### DISCUSSION TITLE
Namespace unification/naming

### DISCUSSION THREAD

**Discussion by michael-hawker** [CommunityToolkit/Windows#32] [Feedback]
_This discussion is for providing feedback on the unified namespace of `CommunityToolkit.WinUI.*` for our components._

Regardless of the package used above, the **namespaces** for code within the packages is now **unified** with a root of `CommunityToolkit.WinUI.*`. This ensures if you target multiple platforms or migrate between them, no code referencing the Toolkit needs to change (just the package)!

This unification helps both library authors and app developers alike!

- **Library authors** can now easily build with XAML on top of Toolkit helpers and components to simultaneously target WinUI 2 and WinUI 3 using the unified namespace as an `xmlns`.
- **App developers** can migrate applications built with vNext Toolkit components between WinUI 2 and 3 or Uno Platform without updating code that references the Toolkit.

Let us know your thoughts on this decision and the naming pattern itself.

**Comment by Pdawg-bytes** [comment]
This is perfect. It will hopefully make migration and upgrading much easier, and it also allows developers to stick with habits and use what they're used to.

**Comment by gus33000** [comment]
This is a very welcome change and one I originally supported years ago when having to target both WinUI 3 and inbox WinUI. There is not many possibilities for developers aside from having conditional compilation references to different xaml files to achieve this without the namespace being identical.

**Comment by michael-hawker** [comment]
Think this decision has been a no brainer, we've clearly seen the benefit while developing and building off of controls to have a unified namespace and what it allows others to build on top of us. Going to close this discussion issue and we'll move forward on the unified namespaces moving forward.
